components/ProtectedRoute.tsx

```diff
@@ -16,21 +15,17 @@ export default function ProtectedRoute({
   const { user, rank } = useSupabase();
   const router = useRouter();
   if (user) {
-    // undefined check is mostly a type guard; if `user`, then `rank` !== undefined
-    if (rank === undefined || rank === Rank.RANK_NULL) {
+    // null check is just a type guard; if `user`, then `rank` !== null
+    if (rank === null || rank === Rank.NEW_USER) {
       // pathname check prevents infinite redirect loop
       if (router.pathname !== "/welcome") {
         return <Redirect route="/welcome" />;
       }
-    } else if (minRank && rankLessThan(rank, minRank)) {
+    } else if (minRank !== undefined && rank < minRank) {
```

Comment:
Wanted to double check `undefined` is the correct thing to check for here -- I think it is b/c it's a prop coming into this component, but with our `null`/`undefined` back-and-forth, wanted to make a note of it

Reply:
Yep, don't wanna do a general truthy check (in case `minRank === 0`), and `minRank` can't be `null` (since that's not a valid rank, so we only allow `Rank | undefined` for `minRank`.

This would be used when we don't specify a `minRank` for a `ProtectedRoute` (e.g. we only want to restrict that page to people who haven't logged in)